woocommerce/woocommerce-ios
# [Product description AI] Show tooltip to introduce feature

Part of: #9975 


## Description

Shows the tooltip to introduce the product description AI feature.

**Changes**
- Show tooltip from `ProductFormViewController`
- Fix the width overflow issue in Tooltip component by injecting available width
- Expose `ButtonTableViewCell`'s button to target the tooltip
- Reset user default values upon logout and switch store

## Testing instructions

WPCOM store
1. Install and login into the store
2. Navigate to the Products tab
3. Open a product, and you can see the tooltip below the "Write with AI" button
4. Tap "Got it" to dismiss the tooltip
5. Go back to product list and open the product again. Validate the the tooltip is not displayed again as you dismissed it
6. Logout and login or switch store
7. Navigate to the Products tab
3. Open a product, and you can see the tooltip below the "Write with AI" button
4. Go back to the product list and open the product again. Do this again and validate that the tooltip is displayed only 3 times. You should not see the tooltip the 4th time you open a product.

Self-hosted store
1. Install and login into the store
2. Navigate to the Products tab
3. Open a product and validate that there is no tooltip visible

**Known issues**
- The table view's header is displayed above the tooltip when we rotate. I tried bringing up the tooltip after viewDidLayout. It doesn't work, though. 
- There are auto layout constraint warnings from the Tooltip component. (This exists in WPiOS as well)

I added both of the above as subtasks. I think it is safe to merge this PR without addressing these issues. I will work on figuring out a fix.

## Screenshots
|  | Light | Dark |
|--------|--------|--------|
| Portriat | ![PL](https://github.com/woocommerce/woocommerce-ios/assets/524475/8e6aa624-0dad-44a0-a86a-5ff643756d92) | ![PD](https://github.com/woocommerce/woocommerce-ios/assets/524475/a6da0738-00d5-4a27-99a4-c0f1853e1794) |
| Landscape | ![LL](https://github.com/woocommerce/woocommerce-ios/assets/524475/3367e6b5-30dd-42f3-b92d-b31028fe7b25) | ![LD](https://github.com/woocommerce/woocommerce-ios/assets/524475/d07d5a56-fcfa-42dd-98c5-8959e4ce4425) | 


---
